fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -973,25 +988,34 @@ Future<std::map<ClusterName, DataClusterMetadata>> listClustersTransaction(Trans
 	return clusters;
 }
 
-ACTOR template <class DB>
-Future<std::map<ClusterName, DataClusterMetadata>> listClusters(Reference<DB> db,
-                                                                ClusterName begin,
-                                                                ClusterName end,
-                                                                int limit) {
-	state Reference<typename DB::TransactionT> tr = db->createTransaction();
-
+ACTOR template <class Transaction>
+Future<std::map<ClusterName, DataClusterMetadata>> listClustersWithTxn(Transaction tr,
+                                                                       ClusterName begin,
+                                                                       ClusterName end,
+                                                                       int limit) {
+	tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+	tr->setOption(FDBTransactionOptions::RAW_ACCESS);
```

Comment:
The difference is that, `listClustersTransaction` does not have a retry loop, while the original `listClusters()` API does. I thought this was intentional. Therefore, I refactored the code a bit so that the retry loop is moved to `listClustersWithTxn()` and the original `listClusters()` calls the former. Within `listClustersTxn()`, it calls `listClustersTransaction()`. The naming is indeed confusing.
If the retry loop is not intentional or not needed, I also vote for removing this `listClustersWithTxn`.

Reply:
The idea with the existing functions is that `listClusters` implements the complete transaction for you, while `listClustersTransaction` is meant to be called in the context of some existing transaction that already has its own retry loop (if required). I think that is appropriate for the new use-case as well, though if we are changing from running a transaction via a DB call to a Transaction call, we should add a retry loop to wherever we are making that call.

Reply:
Done